api/envoy/config/listener/v3/quic_config.proto

```diff
@@ -24,11 +24,25 @@ option (udpa.annotations.file_status).package_version_status = ACTIVE;
 // [#protodoc-title: QUIC listener config]
 
 // Configuration specific to the UDP QUIC listener.
-// [#next-free-field: 12]
+// [#next-free-field: 13]
 message QuicProtocolOptions {
   option (udpa.annotations.versioning).previous_message_type =
       "envoy.api.v2.listener.QuicProtocolOptions";
 
+  // Configuration for UDP cmsg to pass to listener filters via QuicReceivedPacket.
+  // Both level and type must be specified for cmsg to be saved.
+  // More info about CMSG: https://man7.org/linux/man-pages/man3/cmsg.3.html
+  message SaveCmsgConfig {
```

Comment:
This is probably more of a core type (may be used not only for QUIC, but more generic to a socket), so I think the following should be done:
1. it should be moved to a new package in: api/envoy/config/core/v3/.
2. it should probably be renamed to something like `SocketCmsgHeader` as it reflects the structure of a CMSG header (as defined in https://man7.org/linux/man-pages/man3/cmsg.3.html), IIUC.

Reply:
Thanks for the feedback. I moved this and renamed this as suggested, with some tweaks to the comments to make them more generalized.